CliWrap/Exceptions/CommandExecutionException.cs

```diff
@@ -7,48 +7,57 @@ namespace CliWrap.Exceptions
     /// </summary>
     public partial class CommandExecutionException : Exception
     {
+        public ICommandConfiguration Command { get; }
+        public int ExitCode { get; }
+        public string? StandardError { get; }
+
         /// <summary>
         /// Initializes an instance of <see cref="CommandExecutionException"/>.
         /// </summary>
-        public CommandExecutionException(string message) : base(message) {}
-    }
+        public CommandExecutionException(ICommandConfiguration command, int exitCode, string? standardError = null)
+            : base(BuildMessage(command, exitCode, standardError))
+        {
+            Command = command;
+            ExitCode = exitCode;
+            StandardError = standardError;
+        }
 
-    public partial class CommandExecutionException
-    {
-        internal static CommandExecutionException ExitCodeValidation(
-            string filePath,
-            string arguments,
-            int exitCode)
+        private static string BuildMessage(ICommandConfiguration command, int exitCode, string? standardError)
```

Comment:
Can we go back to the factory approach used previously? I know there's a bit of repetition, but I find it more readable.

Reply:
yes, sure